backend/core/src/main/java/org/sonarsource/sonarlint/core/sync/SynchronizationService.java

```diff
@@ -336,6 +340,13 @@ private void synchronizeConnectionAndProjectsIfNeededSync(String connectionId, S
     }
   }
 
+  private void updateConnectionDetails(String connectionId, boolean hasEnterprisePlugin) {
```

Comment:
This method is called only during the sync, which means if a connection is not synchronized, the flag will not be set appropriately and will be false. Have you considered reading from the storage in `shouldUseEnterpriseCSharpAnalyzer` instead of introducing new data in the connection?

Reply:
No, I haven't thought about this possibility, and it sounds good. Let's discuss.